domain/application/state/application.go

```diff
@@ -397,131 +406,144 @@ func (st *ApplicationState) GetUnitUUID(ctx domain.AtomicContext, unitName strin
 	if err != nil {
-		return errors.Trace(err)
+		return "", errors.Trace(err)
 	}
 	nodeUUID, err := uuid.NewUUID()
 	if err != nil {
-		return errors.Trace(err)
+		return "", errors.Trace(err)
 	}
 	createParams := unitDetails{
 		ApplicationID: appID.String(),
 		UnitID:        unitUUID.String(),
 		NetNodeID:     nodeUUID.String(),
 		LifeID:        life.Alive,
 	}
-	if args.PasswordHash != nil {
-		createParams.PasswordHash = *args.PasswordHash
-		createParams.PasswordHashAlgorithmID = 0 //currently we only use sha256
+	if args.Password != nil {
+		createParams.PasswordHash = args.Password.PasswordHash
+		createParams.PasswordHashAlgorithmID = args.Password.HashAlgorithm
 	}
 
 	createUnit := `INSERT INTO unit (*) VALUES ($unitDetails.*)`
 	createUnitStmt, err := st.Prepare(createUnit, createParams)
 	if err != nil {
-		return errors.Trace(err)
+		return "", errors.Trace(err)
 	}
 
 	createNode := `INSERT INTO net_node (uuid) VALUES ($unitDetails.net_node_uuid)`
 	createNodeStmt, err := st.Prepare(createNode, createParams)
 	if err != nil {
-		return errors.Trace(err)
+		return "", errors.Trace(err)
 	}
 
 	createParams.Name = args.UnitName
 
-	if err := tx.Query(ctx, createNodeStmt, createParams).Run(); err != nil {
-		return errors.Annotatef(err, "creating net node row for unit %q", args.UnitName)
-	}
-	if err := tx.Query(ctx, createUnitStmt, createParams).Run(); err != nil {
-		return errors.Annotatef(err, "creating unit row for unit %q", args.UnitName)
-	}
-	if args.CloudContainer != nil {
-		if err := st.upsertUnitCloudContainer(ctx, tx, args.UnitName, nodeUUID.String(), args.CloudContainer); err != nil {
-			return errors.Annotatef(err, "creating cloud container row for unit %q", args.UnitName)
-		}
-	}
-	return nil
-}
-
-// UpdateUnit updates the specified application unit, returning an error
-// satisfying [applicationerrors.UnitNotFoundError] if the unit doesn't exist.
-func (st *ApplicationState) UpdateUnit(
-	ctx domain.AtomicContext, appID coreapplication.ID, args application.UpsertUnitArg,
-) error {
-	err := domain.Run(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		toUpdate, err := st.getUnit(ctx, tx, args.UnitName)
-		if err != nil {
-			return errors.Trace(err)
-		}
-
-		if args.PasswordHash != nil {
-			toUpdate.PasswordHash = *args.PasswordHash
-			toUpdate.PasswordHashAlgorithmID = 0 //currently we only use sha256
-		}
-
-		updateUnit := `
-UPDATE unit SET
-    life_id = $unitDetails.life_id,
-    password_hash = $unitDetails.password_hash,
-    password_hash_algorithm_id = $unitDetails.password_hash_algorithm_id
-WHERE uuid = $unitDetails.uuid
-`
-		updateUnitStmt, err := st.Prepare(updateUnit, *toUpdate)
-		if err != nil {
-			return errors.Trace(err)
+	err = domain.Run(ctx, func(ctx context.Context, tx *sqlair.TX) error {
+		if err := tx.Query(ctx, createNodeStmt, createParams).Run(); err != nil {
+			return errors.Annotatef(err, "creating net node row for unit %q", args.UnitName)
```

Comment:
Do you think row is superfluous here? We  don't care  how it is achieved that we can't achieve it. Maybe I am wrong just asking the question.

Reply:
Sure, I am not wedded to the wording.